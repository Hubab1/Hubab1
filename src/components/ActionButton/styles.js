import {
    css
} from 'emotion';
import styled from '@emotion/styled';

export const root = css`
    border-radius: 21.5px !important;
    height: 45px;
`;

export const label = css`
    text-transform: none;
    font-size: 16px;
`;

export const mainDisabled = css`
    background-color: #D5D8DE !important;
    color: #ffffff !important;
    box-shadow: 0px 1px 5px 0px rgba(0,0,0,0.2), 0px 2px 2px 0px rgba(0,0,0,0.14), 0px 3px 1px -2px rgba(0,0,0,0.12) !important;
`;

export const ButtonContainer = styled.div(props => ({
    marginTop: props.marginTop || 0,
    marginBottom: props.marginBottom || 0,
    textDecoration: 'none',
    display: 'block',
}));
