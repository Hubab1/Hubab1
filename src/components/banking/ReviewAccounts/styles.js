import { css } from 'emotion';
import styled from '@emotion/styled';

export const CardSection = styled.div`
    padding: 15px;
`

export const Card = styled.div`
    margin-bottom: 20px;
    border: 1px solid #EEEEEE;	
    border-radius: 5px;	
    box-shadow: 0 2px 4px 0 rgba(197,197,197,0.5);
    text-align: left;
`

export const totalContainer = css`
    background-color: rgba(86,186,130,0.1);
    display: flex;
    justify-content: space-between;
    padding: 12px 15px;
    width: 100%;
    position: relative;
    left: -15px;
    bottom: -15px;
`

export const incomeEntry = css`
    display: flex;    
    justify-content: space-between;
    margin: 0 0 15px 0;
`