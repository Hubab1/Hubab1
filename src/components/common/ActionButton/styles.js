import {
    css
} from 'emotion';
import styled from '@emotion/styled';

export const root = css`
    border-radius: 21.5px !important;
`

export const label = css`
    text-transform: none;
`

export const ButtonContainer = styled.div(props => ({
    marginTop: props.marginTop || 0,
    marginBottom: props.marginBottom || 0
}))
