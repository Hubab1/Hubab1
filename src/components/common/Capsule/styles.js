import {
    css
} from 'emotion';

export const container = css`
    margin-bottom: 10px;
    a {
        text-decoration: none;
    }
`

export const paperRoot = css`
    box-shadow: 0px 2px 4px 2px rgba(0,0,0,.1) !important;
    border-width: 0px !important;
    padding: 15px !important;
    font-size: 16px !important;
    border-radius: 5px;
    .MuiButton-root {
        font-weight: 600;
    }
`

export const prefix = css`
    font-size: 55px;
    height: 55px;
    line-height: 67px;
    width: 100px;
    text-align: center;
`

export const label = css`
    flex: 1;
    padding-left: 0;
    color: #454B57;
    font-weight: 400;
    display: flex;
    font-size: 16px;
`

export const contentContainer = css`
    display: flex;
    flex: 1;
    align-items: center;
    text-transform: none;
    padding-bottom: 16px;
`

export const buttonRoot = css`
    border-radius: 30px !important;
    text-transform: none !important;
    font-size: 16px !important;
`

export const anchor = css`
    position: relative;
    top: -90px;
`
