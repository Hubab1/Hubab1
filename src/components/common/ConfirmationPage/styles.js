import styled from '@emotion/styled';

export const ImageContainer = styled.div`
    margin-top: 42px;
    margin-bottom: 50px;
`