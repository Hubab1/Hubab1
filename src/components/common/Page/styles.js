import { css } from 'emotion';
import styled from '@emotion/styled';

export const page = css`
  text-align: center;
  margin: auto;
  padding: 10px 0;
`

export const subPage = css`
    padding: 30px 20px;
    max-width: 500px;
    text-align: center;
    margin: auto;
    padding-bottom: 25px;
`

export const Banner = styled.div`
    height: 80px;
    box-shadow: 0 2px 4px 0 rgba(177,181,189,0.4);
    font-size: 30px;
    text-align: center;
    align-items: center;
    display: flex;
    justify-content: center;
`