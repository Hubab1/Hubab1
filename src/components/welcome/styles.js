import styled from '@emotion/styled';


export const BackgroundImage = styled.div( props => ({
    position: 'absolute',
    top: 0,
    left: 0,
    width: '100%',
    height: '100%',
    backgroundPosition: 'center',
    backgroundSize: 'cover',
    zIndex: -5,
    backgroundImage: `url(${props.url})`,
}))

export const BackgroundImageTint = styled.div( props => ({
    position: 'absolute',
    top: 0,
    left: 0,
    width: '100%',
    height: '100%',
    zIndex: -4,
    opacity: .5,
    background: props.primaryColor
}))

export const WelcomeFlexContainer = styled.div`
    position: absolute;
    display: flex;
    flex-direction: column;
    align-content: space-around;
    height: 100%;
    width: 100%;
`

export const WelcomeContainer = styled.div`
    background-color: rgba(0,0,0,0.33);
    text-align: center;
    margin: auto;
    font-family: sans-serif;
    color: #FFFFFF;
    padding: 0 50px 50px 50px;
    max-width: 500px;
`

export const WelcomeLogo = styled.div`
    margin-top: 20px;
    width: 150px;
    margin: auto;
`

export const WelcomeTitle = styled.div`
    font-family: "Proxima Nova";
    font-size: 25px;
    line-height: 30px;
    text-align: center;
    padding-bottom: 20px;
`

export const WelcomeInfo = styled.div`
    color: #FFFFFF;
    font-family: "Proxima Nova";
    font-size: 22px;
    line-height: 30px;
    text-align: center;
`

export const WelcomeInfoTitle = styled.div`
    color: #FFFFFF;
    font-weight: 600;
    font-family: "Proxima Nova";
    font-size: 22px;
    line-height: 30px;
    text-align: center;
    padding-top: 15px;
`

export const WelcomeFooterContainer = styled.div`
    margin: 2% 10%;
    height: 15%;
    text-align: center;
`

export const HomeImageContainer = styled.div`
    position: relative;
    top: -35px;
    width: 30px;
    height: 15px;
    border-top-left-radius: 60px;
    border-top-right-radius: 60px;
    background: inherit;
    text-align: center;
    padding: 10px;
    margin-left: 38%;
`
