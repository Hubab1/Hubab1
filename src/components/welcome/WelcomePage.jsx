import React, { Component, Fragment } from 'react';
import { connect } from 'react-redux';
import { withTheme } from '@material-ui/core/styles';

import funnelImage from '../../assets/images/PoweredByFunnel.png';
import homeImage from '../../assets/images/home-image.png';
import {
    BackgroundImage, BackgroundImageTint, WelcomeFlexContainer, WelcomeContainer,
    WelcomeTitle, WelcomeInfo, WelcomeInfoTitle, WelcomeLogo,
    WelcomeFooterContainer, HomeImageContainer
} from './styles';
import ActionButton from 'components/common/ActionButton';
import history from 'history.js';


export class WelcomePage extends Component {

    renderUnitInfo = (layout, unit_number) => {
        let unitInfo = layout;

        if (unit_number){
            unitInfo = !!unitInfo ? unitInfo.concat(` - ${unit_number}`) : unit_number;
        }
        return unitInfo;
    }

    renderWelcomeInfo = () => {
        const { name, street, city, state, postal_code, layout, unit_number } = this.props.leaseSettings;

        const infoLines = [];

        name && infoLines.push(name);
        street && infoLines.push(street);

        const unitInfo = this.renderUnitInfo(layout, unit_number);
        unitInfo && infoLines.push(unitInfo);

        infoLines.push(`${city}, ${state} ${postal_code}`);

        return (
            infoLines.map( (line, i) => {
                if ( i === 0 ) {
                    return <WelcomeInfoTitle key={i}>{line}</WelcomeInfoTitle>;
                }
                return <WelcomeInfo key={i}>{line}</WelcomeInfo>;
            })
        );
    }

    render() {
        const { client, background_image, logo } = this.props.leaseSettings;

        const helloContent = client ? `Hello ${client.first_name},` : 'Hi There,'

        return (
            <Fragment>
                <BackgroundImage url={background_image} />
                <BackgroundImageTint primaryColor={this.props.theme.palette.primary.main} />
                <WelcomeFlexContainer>
                    <WelcomeLogo>
                        <img src={logo} width="150" alt="company logo"/>
                    </WelcomeLogo>
                    <WelcomeContainer>
                        <HomeImageContainer>
                            <img src={homeImage} width="30" alt="company logo"/>
                        </HomeImageContainer>
                        <WelcomeTitle>
                            {helloContent}
                        </WelcomeTitle>
                        Your new home awaits at
                        <br/>
                        {this.renderWelcomeInfo()}
                    </WelcomeContainer>
                    <WelcomeFooterContainer>
                        <ActionButton
                            onClick={() => history.push('/login')}>Start Application
                        </ActionButton>
                        <img src={funnelImage} width="200" style={{marginTop:'20px'}} alt="funnel logo" />
                    </WelcomeFooterContainer>
                </WelcomeFlexContainer>
            </Fragment>

        );
    }
}

const mapStateToProps = state => ({
    leaseSettings: state.leaseSettings,
});

const connectedWelcomePage = connect(mapStateToProps, null)(WelcomePage);
export default withTheme()(connectedWelcomePage);
