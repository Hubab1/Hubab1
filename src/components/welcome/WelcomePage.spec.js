import React from 'react';
import { shallow } from 'enzyme';

import backgroundImage from 'assets/images/desktop_cortlandimg.png';
import logoImage from 'assets/images/cortland_logo.png';
import { WelcomePage } from './WelcomePage';
import { WelcomeTextContainer } from './styles';

let defaultProps;
beforeEach(() => {
    defaultProps = {
        configuration: {
            application_fee: null,
            background: backgroundImage,
            community: {
                building_name: "The Excelsior",
                building_number: "601",
                city: "New York",
                normalized_street_address: "601 W 57TH ST",
                postal_code: "10019",
                state: "NY",
            },
            id: 1,
            logo: logoImage,
            primary_color: "286165",
            rental_options_config: {
                guarantor: {
                    limit: 1
                },
                pets: {
                    limit: 1
                },
                roommates:{
                    limit: 3
                }
            },
            secondary_color: "FFFFFF",
            client: {
                person: {
                    email: "callieapi@example.com",
                    first_name: "Callie",
                    id: 280066,
                    last_name: "Api",
                    phone_1: "1234543123",
                }
            },
            unit: {
                id: 2747820,
                unit_number: "4B"
            }
        },
        theme: {palette:
            {primary:
                {main: "rgba(40,97,101,1)"}
            }
        }
    };
})

it('renders Hello <client name> when client is included in configuration', function(){
    const wrapper = shallow(<WelcomePage {...defaultProps}/>);

    expect(wrapper.text().includes('Hello Callie,')).toBeTruthy();
});

it('renders Hi There, when client is not included in configuration', function(){
    delete defaultProps.configuration.client;
    const wrapper = shallow(<WelcomePage {...defaultProps}/>);

    expect(wrapper.text().includes('Hi There,')).toBeTruthy();
});

it('renders correct info when all client and unit info are passed', () => {
    const wrapper = shallow(<WelcomePage {...defaultProps}/>);

    expect(wrapper.find(WelcomeTextContainer).text()).toContain('Excelsior');
    expect(wrapper.find(WelcomeTextContainer).text()).toContain('601 W 57TH ST');
    expect(wrapper.find(WelcomeTextContainer).text()).toContain('New York, NY 10019');
    expect(wrapper.find(WelcomeTextContainer).text()).toContain('Unit 4B');
});

it('renders correct info when all client and unit info is omitted', () => {
    delete defaultProps.configuration.unit;

    const wrapper = shallow(<WelcomePage {...defaultProps}/>);

    expect(wrapper.find(WelcomeTextContainer).text()).toContain('Excelsior');
    expect(wrapper.find(WelcomeTextContainer).text()).toContain('601 W 57TH ST');
    expect(wrapper.find(WelcomeTextContainer).text()).toContain('New York, NY 10019');
    expect(wrapper.find(WelcomeTextContainer).text()).not.toContain('Unit 4B');

});
