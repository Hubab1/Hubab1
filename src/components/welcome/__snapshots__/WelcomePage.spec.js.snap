// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot for welcome page 1`] = `
Array [
  <div
    className="css-7gulox"
  />,
  <div
    className="css-1u4ctym"
  />,
  <div
    className="css-1cs06h0"
  >
    <div
      className="css-1wiy7ua"
    >
      <img
        alt="company logo"
        src="cortland_logo.png"
        width="150"
      />
    </div>
    <div
      className="css-1l6arv8"
    >
      <div
        className="css-1kbcrrf"
      >
        <img
          alt="company logo"
          src="home-image.png"
          width="30"
        />
      </div>
      <div
        className="css-v8g9jw"
      >
        Hello Chester,
      </div>
      Your new home awaits at
      <br />
      <div
        className="css-enz7gz"
      >
        Cortland
      </div>
      <div
        className="css-1mlti9w"
      >
        2173 Highland Ave S
      </div>
      <div
        className="css-1mlti9w"
      >
        1 Bedroom - 3B
      </div>
      <div
        className="css-1mlti9w"
      >
        Birmingham, AL 35205
      </div>
    </div>
    <div
      className="css-q38i9v"
    >
      <button
        className="MuiButtonBase-root-27 MuiButton-root-1 css-xrj6rj MuiButton-contained-12 MuiButton-containedPrimary-13 MuiButton-raised-15 MuiButton-raisedPrimary-16 MuiButton-fullWidth-26"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="submit"
      >
        <span
          className="MuiButton-label-2 css-1tzl5gn"
        >
          Start Application
        </span>
        <span
          className="MuiTouchRipple-root-30"
        />
      </button>
      <img
        alt="funnel logo"
        src="PoweredByFunnel.png"
        style={
          Object {
            "marginTop": "20px",
          }
        }
        width="200"
      />
    </div>
  </div>,
]
`;
